src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/View/Controller/QRScanner/ExposureSubmissionQRScannerViewModel.swift

```diff
@@ -47,31 +49,99 @@ class ExposureSubmissionQRScannerViewModel: NSObject, AVCaptureMetadataOutputObj
 
 	// MARK: - Internal
 
-	let onError: (QRScannerError, _ reactivateScanning: @escaping () -> Void) -> Void
-	let onCancel: () -> Void
+	enum TorchMode {
+		case notAvailable
+		case lightOn
+		case ligthOff
+	}
 
-	func activateScanning() {
-		isScanningActivated = true
+	let captureSession: AVCaptureSession
+
+	func startCaptureSession() {
+		switch AVCaptureDevice.authorizationStatus(for: .video) {
+		case .authorized:
+			Log.info("AVCaptureDevice.authorized - enable qr code scanner")
+			activateScanning()
+		case .notDetermined:
+			AVCaptureDevice.requestAccess(for: .video) { [weak self] isAllowed in
+				guard isAllowed else {
+					self?.onError(.cameraPermissionDenied) {
+						Log.error("camera requestAccess denied - stop here we can't go on", log: .ui)
+					}
+					return
+				}
+				self?.activateScanning()
+			}
+		default:
+			onError(.cameraPermissionDenied) {
+				Log.info(".cameraPermissionDenied - stop here we can't go on", log: .ui)
```

Comment:
Shouldn't we show something to the user here, so he knows, he needs to give access?